core/packages/contracts/src/Vault.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+import "openzeppelin/access/AccessControl.sol";
+import "./IVault.sol";
+
+contract Vault is IVault, AccessControl {
+    event Deposited(bytes indexed sovereign, uint256 amount);
+    event Withdrawn(bytes indexed sovereign, address recipient, uint256 amount);
+
+    error InsufficientBalance();
+    error ZeroAmount();
+    error CannotSendFunds();
+
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant WITHDRAW_ROLE = keccak256("WITHDRAW_ROLE");
+
+    // Mapping of sovereign to balance
+    mapping(bytes => uint256) public balances;
+
+    constructor() {
+        _grantRole(ADMIN_ROLE, msg.sender);
+        _setRoleAdmin(WITHDRAW_ROLE, ADMIN_ROLE);
+    }
+
+    receive() external payable {
+        revert("Must use deposit function");
+    }
+
+    function deposit(bytes calldata sovereign) external payable {
```

Comment:
- No DEPOSIT_ROLE, anyone can deposit? Is that okay? Also, what is this function used for? Looks used just to collect fees? Should add comment/description

Reply:
Yes, this is OK, anyone can deposit funds here, by design.

Parachain teams must periodically deposit funds to cover relayer rewards.

These funds can be raised by selling DOT from their sovereign account on BridgeHub. This account contains funds charged to users for sending messages across the bridge to Ethereum.

So in the BridgeHub->Ethereum direction, parachain teams control two sovereign accounts:
* On BridgeHub, funds collected by users for using the bridge
* On Ethereum, funds rewarded to message relayers.